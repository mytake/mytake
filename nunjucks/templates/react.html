{% extends "base.html" %}

{% block endbody %}
  <!-- Dependencies -->
  <script src="https://unpkg.com/react@15/dist/react.js"></script>
  <script src="https://unpkg.com/react-dom@15/dist/react-dom.js"></script>

  <!-- Main -->
  <script src="./bundle-0.0.3.js"></script>
{% endblock %}
