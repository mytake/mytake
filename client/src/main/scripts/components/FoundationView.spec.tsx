import * as React from "react";
import * as renderer from "react-test-renderer";
import FoundationView from "./FoundationView";

jest.mock("./TimelineView", () => ({
  default: "TimelineView"
}));

test("With fact in URL", () => {
  const path = "/foundation/bill-of-rights";
  const tree = renderer
    .create(<FoundationView path={path} search={""} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Without fact in URL", () => {
  const path = "/foundation";
  const hashURL = "";
  const tree = renderer
    .create(<FoundationView path={path} search={""} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
