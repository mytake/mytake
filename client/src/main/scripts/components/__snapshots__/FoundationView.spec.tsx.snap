// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With fact in URL 1`] = `
<div
  className="foundation"
>
  <div>
    <div
      className="foundation__inner-container"
    >
      <h1
        className="foundation__heading"
      >
        Foundation of Facts
      </h1>
      <p
        className="timeline__instructions"
      >
        Explore Facts in the timeline below.
      </p>
    </div>
    <TimelineView
      path="/foundation/bill-of-rights"
    />
  </div>
</div>
`;

exports[`Without fact in URL 1`] = `
<div
  className="foundation"
>
  <div>
    <div
      className="foundation__inner-container"
    >
      <h1
        className="foundation__heading"
      >
        Foundation of Facts
      </h1>
      <p
        className="timeline__instructions"
      >
        Explore Facts in the timeline below.
      </p>
    </div>
    <TimelineView
      path="/foundation"
    />
  </div>
</div>
`;
